src/router.rs

```diff
@@ -0,0 +1,84 @@
+use crate::{
+    config::Configuration,
+    routes::{health_check, zome_call},
+    service::AppState,
+};
+use axum::{routing::get, Router};
+
+pub fn hc_http_gateway_router(configuration: Configuration) -> Router {
```

Comment:
Following the [example how to test the router](https://github.com/tokio-rs/axum/blob/main/examples/testing/src/main.rs#L37), this is a builder for the router which can then be tested.

Reply:
I think we should still be able to test path extraction and validation separately. So I suggest to make the app connection pool mockable? Then we can test with state but don't need a Holochain connection.

Reply:
Sounds good, we'll figure that out as we integrate these PRs